mock/mock.go

```diff
@@ -500,7 +500,9 @@ func (m *Mock) AssertCalled(t TestingT, methodName string, arguments ...interfac
 	if !m.methodWasCalled(methodName, arguments) {
 		var calledWithArgs []string
 		for _, call := range m.calls() {
-			calledWithArgs = append(calledWithArgs, fmt.Sprintf("%v", call.Arguments))
+			if call.Method == methodName {
```

Comment:
Hmm. Would we have false information here if two methods in the call stack shared the same name but from different packages? i.e.

pkg1.FuncA() => pkg2.FuncB() => pkg2.FuncA()

Reply:
I don't think so, because It would be another mock instance.

Reply:
Fair enough.

Could we add unit tests for this new logic? I think it would make sense if we at least cover the use-case mentioned in the PR description as well as the same-function name one I mentioned above. Other cases would also be welcome :)